README.md

```diff
@@ -0,0 +1,453 @@
+
+// feeding your query to `observe` will give you an observable on the query
+// N.B. until now no fetch is yet attempted, avenger will wait until the first subscription is issued
+const observable: Observable<QueryResult<Error, User>> = observe(userQuery);
+
+// after this the fetch function triggers
+observable.subscribe(dispatchError, setCurrentUser);
+
+// alternatively you can call `run` on your query and it will return a TaskEither<Error, User>
+// you can then use it imperatively
+const task: TaskEither<Error, User> = userQuery.run(1);
+const result: Either<Error, User> = await task.run();
+```
+
+altough the `run` method is available to check a query result imperatively, it is highly suggested the use of the `observe` utility in order to be notified in real time of when data changes.
+
+Either way, whenever you ask for a query result you will end up with an object with the [**`QueryResult`**](#QueryResult) signature that conveniently lets you fold to decide the best way to handle the result. The `fold` method takes 3 functions as parameters, the first is used to handle a `Loading` result, the second is used on case a `Failure` occurs and the last one handles `Success` values.
+
+
+## composing queries
+You can build bigger queries from smaller ones in two ways:
+
+- by composing them with [**`compose`**](#compose): when you need your queries to be sequentially run with the results of one feeding the other, you can use `compose`:
+
+```ts
+import { compose } from 'avenger/lib/Query';
+
+type UserPreferences = { color: string };
+
+// note that the two ends of the composed functions must have compatible types
+declare function getUser(userId: number): TaskEither<Error, User>;
+declare function getUserPreferences(
+  user: User
+): TaskEither<Error, UserPreferences>;
+
+const userQuery: ObservableQuery<number, Error, User> = queryShallow(
+  getUser,
+  refetch
+);
+
+const preferencesQuery: ObservableQuery<
+  User,
+  Error,
+  UserPreferences
+> = queryShallow(getUserPreferences, refetch);
+
+const composition: Composition<number, Error, UserPreferences> = compose(
+  userQuery,
+  preferencesQuery
+);
+```
+
+- by grouping them with [**`product`**](#product): when you don't need to run the queries sequentially but would like to conveniently group them and treat them as if they where one you can use `product`:
+
+```ts
+const grouped = product({ myQuery, myQuery2 });
+```
+
+# commands
+Since now we only described how fetching of data is performed, when you need to update / insert data remotely you can make use of [**`command`**](#command):
+
+```ts
+declare function updateUserPreferences({
+  color: string
+}): TaskEither<Error, void>;
+
+const updatePreferencesCommand = command(updateUserPreferences, {
+  preferencesQuery
+});
+```
+
+`command` accepts `Fetch` function that will be used to modify the remote data source and a second parameter, a `Record` of `query`es that will be invalidated once the command is succesfully run:
```

Comment:
So the `Fetch` concept is not strictly related to fetching some data, but to a more general remote call. Is it because of widely used terminology in the frontend?

Reply:
you are correct, `Fetch` is not a super descriptive name indeed... cc @giogonzo

Reply:
In fp-ts lingo this is just a `ReaderTaskEither`: 3 effects (function/input, async, error) stacked. `fetch` is just the old avenger lingo, introduced a few years ago when describing the API for the first time with Giulio (see https://github.com/buildo/avenger#fetch), in any case not something particularly meaningful or essential.

My opinion at the moment is: when this is backported to fp-ts v2 (should happen soon), creating and handling `ReaderTaskEither`s is going to be cheaper and more intuitive (no `class` wrapping), so the Fetch name could just be dropped in favor of `ReaderTaskEither`.

Do you have any suggestions in the meantime?

Regarding:
> So the Fetch concept is not strictly related to fetching some data, but to a more general remote call

I'd say not even to the "remote call" part. This is probably not obvious from these docs, but a query is characterized by one or more of the 3 effects described above. E.g. we use it to access `localStorage`, which is only effectful in the sense of IO (sync) and possibly Either (validation of the contents), but definitely not "remote"